include/mbgl/style/layer.hpp

```diff
@@ -117,14 +111,12 @@ class Layer : public mbgl::util::noncopyable {
     void setMaxZoom(float) const;
 
     // Private implementation
-    const std::unique_ptr<Impl> baseImpl;
+    const std::shared_ptr<Impl> baseImpl;
```

Comment:
@jfirebaugh I figured this must be the reason that all the property-function render tests are failing. It makes me think it's because there is some state now not copied over / held on to when creating the buckets (since that is the main change).

If I understand correctly; in [the painter](https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/renderer/painter_circle.cpp#L15) the paint property binders (created in the bucket from the copied paint properties) and the evaluated paint properties form the `Layer::Impl` come together and then in the program the current properties are bound to attributes and interpolated from the values the binders were originally created with right?